src/tools/PerfLabGenericEventSourceLTTngProvider/build.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+while getopts ":o:" opt; do
+    case $opt in
+        o)
+            OUTPUT=$OPTARG
+            ;;
+        :)
+            echo "Option -$OPTARG requires an argument." >&2
+            exit 1
+            ;;
+        \?)
+            echo "Invalid option: -$OPTARG." >&2
+            exit 1
+            ;;
+    esac
+done
+
+pushd $(dirname $(realpath $0)) > /dev/null
+
+# install liblttng-ust-dev
+
+gcc -c -I. -fpic PerfLabGenericEventSourceLTTngProvider.c && \
+gcc -o $OUTPUT/PerfLabGenericEventSourceLTTngProvider.so -shared PerfLabGenericEventSourceLTTngProvider.o -llttng-ust
```

Comment:
Do you know what the default optimization level is for this compilation?  I did a bit of searching and didn't find a clear answer.  Mostly just want to make sure that we're not trying to run "debug" codegen unintentionally.

Reply:
Good catch! I added `O2` optimization level.